frontend/src/scenes/funnels/funnelLogic.ts

```diff
@@ -574,6 +574,10 @@ export const funnelLogic = kea<funnelLogicType>({
                     actions.loadPeople(values.stepsWithCount)
                 }
             }
+            if (!insightLogic.values.insight.id) {
+                actions.createInsight(values.filters)
+            }
+            insightLogic.actions.updateInsightFilters(values.filters)
```

Comment:
This `actions.createInsight` is actually `insightHistoryLogic.actions.createInsight`, which calls `api.create('api/insight')`. The problem is that `funnelLogic` is also mounted for each dashboard item, so this makes a lot of mess when loading a dashboard:

![image](https://user-images.githubusercontent.com/53387/133627357-85dde849-c8e7-4760-9f52-1fae07757576.png)

You can just `if (!props.dashboardItemId)` all the code that touches `insightLogic` as a quick solution.

Reply:
good catch!!! I've feature flagged this (since it's a saved insights flow functionality) and also made sure to update only when an insight actually exists to be updated